otri/database/postgresql_adapter.py

```diff
@@ -4,160 +4,101 @@
+        time. If you need to fetch few rows at a time but do not need a `Stream` object use
+        sqlalchemy's `yield_per(amount)` in a session.
+        A new connection is opened for each stream.
 
         Parameters:
-            query : DatabaseQuery
-                Executes the query on the given query.category and the given query.filters
-            batch_size : int
-                The number of rows the database should load before making them available.
-                The iterable still always yields a single item.
+            query\n
+                Query to run, must be an sqlalchemy object, must be a read only query.\n A new
+                connection is opened in order to run it, so if you got the query from a preexisting
+                session, you don't need to keep it open.\n
+            batch_size : int\n
+                The number of rows the database should load before making them available.\n
+                The iterable still always yields a single item.\n
         Returns:
             An Iterable stream of database rows that match the query.
         '''
-        return PostgreSQLStream(self.connection, query, batch_size)
+        if not isinstance(query, Query):
+            raise ValueError("Not an SQLAlchemy query.")
 
-    def __create_table(self, table_name: str):
-        '''
-        Crates a new table, if it doesn't already exist.
+        query = query.statement.compile(self._engine, compile_kwargs={"literal_binds": True}).string
+        return PostgreSQLStream(self._engine.raw_connection(), query, batch_size)
 
-        Parameters:
-            table_name : str
-                Name of the table to create.
+    def _connection_string(self):
         '''
-        log.i("creating new pgSQL DB table {}".format(table_name))
-        self.cursor.execute(
-            "CREATE TABLE {} (id BIGSERIAL PRIMARY KEY, data_json JSON NOT NULL);".format(table_name))
-        self.connection.commit()
-
-    def __table_exists(self, table_name: str):
+        Returns:\n
+            Connection string for a PostgreSQL database using psycopg2 driver.
+            Format fields are, in order: user, password, host, port, database.
         '''
-        Determines if the given table exists.
-
-        Parameters:
-            table_name : str
-                Table name to check the existance of.
-        Returns:
-            True if table exists, False otherwise.
-        '''
-        self.cursor.execute(
-            "SELECT tablename FROM pg_catalog.pg_tables WHERE tablename = %s;", (table_name,))
-        return len(self.cursor.fetchall()) > 0
-
-    def close(self):
-        '''
-        Closes database connection.
-        '''
-        if(self.connection):
-            log.i("closing pgSQL DB connection")
-            self.cursor.close()
-            self.connection.close()
+        return "postgresql+psycopg2://{}:{}@{}:{}/{}"
```

Comment:
Is the `postgresql+psycopg2://` defined by sqlalchemy or what? Where can I find other examples of connection strings?

Reply:
Sadly yeah it's sqlalchemy specific. It's usually `database+driver://...`